biodatasets/mediqa_nli/mediqa_nli.py

```diff
@@ -0,0 +1,181 @@
+# coding=utf-8
+# Copyright 2022 The HuggingFace Datasets Authors and the current dataset script contributor.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Natural Language Inference (NLI) is the task of determining whether a given hypothesis can be
+inferred from a given premise. Also known as Recognizing Textual Entailment (RTE), this task has
+enjoyed popularity among researchers for some time. However, almost all datasets for this task
+focused on open domain data such as as news texts, blogs, and so on. To address this gap, the MedNLI
+dataset was created for language inference in the medical domain. MedNLI is a derived dataset with
+data sourced from MIMIC-III v1.4. In order to stimulate research for this problem, a shared task on
+Medical Inference and Question Answering (MEDIQA) was organized at the workshop for biomedical
+natural language processing (BioNLP) 2019. The dataset provided herein is a test set of 405 premise
+hypothesis pairs for the NLI challenge in the MEDIQA shared task. Participants of the shared task
+are expected to use the MedNLI data for development of their models and this dataset was used as an
+unseen dataset for scoring each participant submission.
+
+This script expects the filepath to the zip archive containing the MEDIQA NLI Dataset on the users
+local machine, e.g. local/path/to/mednli-for-shared-task-at-acl-bionlp-2019-1.0.1.zip for version
+1.0.1, that is passed to `datasets.load_datset` via the `data_dir` kwarg. This loader script will
+read the archive files directly (i.e. the user should not uncompress, untar or unzip any of the
+files). The uncompressed directory should contain the following files:
+
+mednli-for-shared-task-at-acl-bionlp-2019-1.0.1
+├── mednli_bionlp19_shared_task.jsonl
+├── mednli_bionlp19_shared_task_ground_truth.csv
```

Comment:
Description of local file(s).

Reply:
sorry I wasn't clear ... suppose you have the directory structure, 
```
/home/user/data/mediqa_nli_data
|--mednli-for-shared-task-at-acl-bionlp-2019-1.0.1.zip
```
then the command below should work w/o the user manually unzipping the file 
`ds = load_dataset("/path/to/loader_script.py", name=..., data_dir=/home/user/data/mediqa_nli_data)`

Reply:
yes ... in this way you could imagine a common pattern among all local datasets, 

* create a directory to store data for that task/dataset
* download all the compressed files for that dataset and put them in the directory above
* pass the directory above to `load_dataset` via the `data_dir` directory

for datasets like this one (with only one compressed file), you could imagine doing it like you have done (with `data_dir` pointing directly to the zip file) but, 
a) its a little strange to pass a file to an argument called `data_dir`
b) its a pattern that does not generalize to datasets that contain more than one compressed file.

Reply:
Gotcha! Okay my latest commit should address this. It now expects the path to a parent directory that contains the zip archive.